### PR TITLE
Specify guest accounts

### PR DESCRIPTION
**Pull request by illicitonion**

### CODE REVIEW

File: api/client-server/v2_alpha/registration.yaml [hunk 1]
```diff
@@ -17,7 +17,24 @@ paths:
       summary: Register for an account on this homeserver.
       description: |-
         Register for an account on this homeserver.
+
+        There are two kinds of user account:
+
+        - user accounts may use the full API described in this specification.
+
+        - guest accounts may have severely limited permissions, and also may not be supported by all servers.
```
Comment (Kegsay) [inline api/client-server/v2_alpha/registration.yaml]: Wording feels clunky here. "There are two kinds of user account: user accounts [..]". Perhaps:

```
There are two kinds of account:

- "User" accounts. These accounts may use the full API described in this specification.
- "Guest" accounts. These accounts may have limited permissions and may not be supported by all servers.
```

File: api/client-server/v2_alpha/registration.yaml [hunk 2]
```diff
@@ -17,7 +17,24 @@ paths:
...
+        - in: query
+          name: kind
+          type: string
+          x-example: guest
+          required: false
+          default: user
+          enum:
+            - guest
+            - user
+          description: The kind of account to register. Defaults to user.
```
Comment (Kegsay) [inline api/client-server/v2_alpha/registration.yaml]: Backticks on `user` please :angel:

**Comment by Kegsay** [comment]
LGTM; I am assuming that this PR will be followed by more which will detail which APIs "Guest" accounts have access to. Since this is a general concept applicable to all APIs, we should probably include an `x-account-kind` field or something so we can say which APIs are accessible by "Guest" account.

File: api/client-server/v2_alpha/registration.yaml (hunk 1 above)
Comment (illicitonion) [inline api/client-server/v2_alpha/registration.yaml]: Done. Used lower case `user` and `guest`, and used backticks not double-quotes.

File: api/client-server/v2_alpha/registration.yaml (hunk 2 above)
Comment (illicitonion) [inline api/client-server/v2_alpha/registration.yaml]: Done

**Comment by illicitonion** [comment]
Something like that sounds plausible, for sure :)

**Comment by illicitonion** [comment]
Manually merged (yay squash!)

**Pull request closed without merging**